--- a/test/support/os_helper.py
+++ b/test/support/os_helper.py
@@ -19,7 +19,9 @@
 
 # Disambiguate TESTFN for parallel testing, while letting it remain a valid
 # module name.
-TESTFN_ASCII = "{}_{}_tmp".format(TESTFN_ASCII, os.getpid())
+import random as _random
+TESTFN_ASCII = "{}_{}_tmp".format(TESTFN_ASCII, _random.randint(0, 1_000_000))
+del _random
 
 # TESTFN_UNICODE is a non-ascii filename
 TESTFN_UNICODE = TESTFN_ASCII + "-\xe0\xf2\u0258\u0141\u011f"
