--- a/test/libregrtest/utils.py
+++ b/test/libregrtest/utils.py
@@ -451,7 +451,4 @@
     # testing (see the -j option).
     # Emscripten and WASI have stubbed getpid(), Emscripten has only
     # millisecond clock resolution. Use randint() instead.
-    if support.is_emscripten or support.is_wasi:
-        nounce = random.randint(0, 1_000_000)
-    else:
-        nounce = os.getpid()
+    nounce = random.randint(0, 1_000_000)
